### PR TITLE
Symlink 

### PR DESCRIPTION
**Pull request by jomonkj**

### CODE REVIEW

**Comment by jomonkj** [comment]
consider if it run from docker.
will produce 
```ssh
bash: sudo: command not found
```
 i removed sudo.

https://github.com/hiqdev/hidev/blob/master/src/helpers/FileHelper.php#L104

File: src/components/Directory.php [hunk 1]
```diff
@@ -22,11 +22,16 @@ public function save()
     {
         FileHelper::mkdir($this->_path);
 
-        foreach ($this->getItems() as $id => $config) {
-            $type = isset($config['template']) || isset($config['copy']) ? 'File' : 'Directory';
+       foreach ($this->getItems() as $id => $config) {
+            $type = 'Directory';
+            if (isset($config['template']) || isset($config['copy'])) {
+                $type = 'File';
+            } elseif (isset($config['dest'])) {
```
Comment (hiqsol) [inline src/components/Directory.php]: Please rename dest -> symlink

File: src/components/Symlink.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @link      https://github.com/hiqdev/hidev
+ * @package   hidev
+ * @license   BSD-3-Clause
+ * @copyright Copyright (c) 2015-2018, HiQDev (http://hiqdev.com/)
+ */
+
+namespace hidev\components;
+
+use hidev\base\File as FileObj;
```
Comment (hiqsol) [inline src/components/Symlink.php]: Remove unused `use`s.

File: src/components/Symlink.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Symlink component.
+ *
+ * @author Jomon Johnson <iamjomonjohnson@gmail.com>
+ */
+class Symlink extends File
+{
+    /**
+     * @var list destimation
+     */
+    public $dest;
```
Comment (hiqsol) [inline src/components/Symlink.php]: dest -> symlink

**Review by hiqsol — CHANGES_REQUESTED** [review]

**Comment by hiqsol** [comment]
Does it do what you need?

**Comment by jomonkj** [comment]
working but showing an error like
```
Couldn't read file: ./public/admin/media
```

**Comment by jomonkj** [comment]
waiting for merge :smile:

File: src/helpers/FileHelper.php [hunk 4]
```diff
@@ -98,6 +98,10 @@ public static function mkdir($path)
     public static function symlink($src, $dst)
     {
         try {
+            if(is_link($dst)) {
+                unlink($dst);
```
Comment (hiqsol) [inline src/helpers/FileHelper.php:103]: this is not efficient
better readlink and compare with desired dst and then delete or do nothing

**Review by hiqsol — CHANGES_REQUESTED** [review]

**Comment by jomonkj** [comment]
Done :smile:

**Pull request merged**